internal/backend/httpsvc/metrics.go

```diff
@@ -0,0 +1,78 @@
+package httpsvc
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"go.autokitteh.dev/autokitteh/internal/backend/telemetry"
+	"go.opentelemetry.io/otel/metric"
+)
+
+type apiMetrics struct {
+	counter  metric.Int64Counter
+	duration metric.Int64Histogram
+}
+
+var metrics sync.Map
+
+// metrics    map[string]apiMetrics
+// metricsMux sync.Mutex
+
+func getServiceAPIMetrics(serviceAPI string, t *telemetry.Telemetry) (*apiMetrics, error) {
+	if m, ok := metrics.Load(serviceAPI); ok {
+		return m.(*apiMetrics), nil
+	}
+
+	cntName := fmt.Sprintf("api.%s", serviceAPI)
+	histName := fmt.Sprintf("api.%s.duration", serviceAPI)
+
+	counter, err := t.NewCounter(cntName, fmt.Sprintf("GRPC request counter (%s)", cntName))
+	if err != nil {
+		return nil, err
+	}
+	histogram, err := t.NewHistogram(histName, fmt.Sprintf("GRPC request duration (%s)", histName))
+	if err != nil {
+		return nil, err
+	}
+	m := apiMetrics{counter, histogram}
+	metrics.LoadOrStore(serviceAPI, &m)
```

Comment:
why not just `Store`?

Reply:
ah, you don't want to overwrite in case something else called get in the meanwhile. I would comment about it.

Reply:
Otherwise I need to protect this in mutex as well, yes.. and we decided to use sync.Map only because of that. Otherwise (in pure read access, after the counter was added to the map) we don't need to do anything, since OTEL code is handling sync on counter update